### DISCUSSION TITLE
Sluggish Performance of LSP auto completion pop up

### DISCUSSION THREAD

**Discussion by that-ambuj** [helix-editor/helix#8351] [General] [answered]
Aside from the lack of customization with helix's LSP completion pop-up, when `auto-completion = true` is set(default). There is a 1-2 seconds of debounce to *pop-up*. It's not very instant and fast like neovim's nvim-cmp. I don't know if this is a design decision or just poor performance optimization but the problem is there irrespective of the LSP or Language used(Go and Rust in my case). It there some way we can improve this behaviour?

**Comment by that-ambuj** [comment]
I have also set `completion-trigger-len` to 0 but the performance is still bad.

**Comment by kirawi** [comment, marked as answer]
This is adjusted by `idle-timeout`

**Reply by that-ambuj** [reply]
Thanks! In my opinion 400ms of idle-timeout with 2 character delay for auto completion is a bit too slow for being the default. I think we should change the default to 200-250ms.

I also realised how ambiguous this option is and hard to find. Maybe, this option should renamed, what do you think?

**Reply by kirawi** [reply]
On master it was reduced to `250` ms. Probably not worth breaking configs for a name change like you're suggesting though.

**Reply by that-ambuj** [reply]
Renaming shouldn't be a breaking change, we can use aliasing and slowly deprecate the old name but maybe that's a bit too extreme. Or maybe it's just me who finds the naming confusing 😅.

P.S. I think a better name would be `completion-debounce`

**Reply by that-ambuj** [reply]
Thanks again, you've been a great help!

**Reply by kirawi** [reply]
`idle-timeout` is used for other situations like syntax highlighting in pickers. It could be split off, but IIRC it wasn't because it would increase code complexity for not a lot of benefit. We also don't want aliasing; it's just kind of unnecessary and would make things confusing. I think the current idea is to hold off on any breaking changes to the config along those lines until `config.toml` is replaced by the plugin system.

**Reply by that-ambuj** [reply]
Fair enough, we should try to avoid any breaking changes if possible. Maybe, updating the docs to be less ambiguous about this option would be a better alternative?
